contracts/staking/contracts/src/vaults/StakingPoolRewardVault.sol

```diff
@@ -77,265 +57,55 @@ contract StakingPoolRewardVault is
-    /// which is enforced by the staking contract. We do not enforce it here to save (a lot of) gas.
-    /// Note that this is only callable by the staking contract, and when
-    /// not in catastrophic failure mode.
-    /// @param poolId Unique Id of pool.
-    /// @param amount Amount in ETH to record.
-    /// @param operatorOnly Only attribute amount to operator.
-    /// @return operatorPortion Portion of amount attributed to the operator.
-    /// @return membersPortion Portion of amount attributed to the pool.
-    function recordDepositFor(
-        bytes32 poolId,
-        uint256 amount,
-        bool operatorOnly
-    )
-        external
-        onlyStakingContract
-        returns (
-            uint256 operatorPortion,
-            uint256 membersPortion
-        )
-    {
-        // update balance of pool
-        (operatorPortion, membersPortion) = _incrementPoolBalances(poolById[poolId], amount, operatorOnly);
-        return (operatorPortion, membersPortion);
-    }
-
-    /// @dev Withdraw some amount in ETH of an operator's reward.
-    /// Note that this is only callable by the staking contract, and when
-    /// not in catastrophic failure mode.
-    /// @param poolId Unique Id of pool.
-    function transferOperatorBalanceToEthVault(
-        bytes32 poolId,
-        address operator,
-        uint256 amount
-    )
+    /// @dev Deposit an amount of ETH (`msg.value`) for `poolId` into the vault.
+    /// Note that this is only callable by the staking contract.
+    /// @param poolId that holds the ETH.
+    function depositFor(bytes32 poolId)
         external
+        payable
         onlyStakingContract
     {
-        if (amount == 0) {
-            return;
-        }
-
-        // sanity check - sufficient balance?
-        uint256 operatorBalance = uint256(poolById[poolId].operatorBalance);
-        if (amount > operatorBalance) {
-            LibRichErrors.rrevert(LibStakingRichErrors.AmountExceedsBalanceOfPoolError(
-                amount,
-                poolById[poolId].operatorBalance
-            ));
-        }
-
-        // update balance and transfer `amount` in ETH to staking contract
-        poolById[poolId].operatorBalance = operatorBalance.safeSub(amount).downcastToUint96();
-        _transferToEthVault(operator, amount);
-
-        // notify
-        emit RewardWithdrawnForOperator(poolId, amount);
+        balanceByPoolId[poolId] = balanceByPoolId[poolId].safeAdd(msg.value);
+        emit EthDepositedIntoVault(msg.sender, poolId, msg.value);
     }
 
     /// @dev Withdraw some amount in ETH of a pool member.
-    /// Note that this is only callable by the staking contract, and when
-    /// not in catastrophic failure mode.
+    /// Note that this is only callable by the staking contract.
     /// @param poolId Unique Id of pool.
     /// @param amount Amount in ETH to transfer.
-    function transferMemberBalanceToEthVault(
+    function transferToEthVault(
         bytes32 poolId,
```

Comment:
What do we think about just passing in the `ethVault` address here? Then we wouldn't have to set up `_ethVault`, etc.

Reply:
I like it a lot. Then they can't get out of sync either.